### DISCUSSION TITLE
Turn off and reconnect to Wi-Fi while real-time websocket connection is active, then unable to reach the server

### DISCUSSION THREAD

**Discussion by tamas9804** [miguelgrinberg/microdot#272] [Q&A]
Hi miguelgrinberg, 

I am working on a home project where I send data to a website and then the website sends back an "ok" message via websocket every 2 sec. If I just send data and I don't get back anything, the website dies after 3-4 data sending in iOS. My problem: I can't reconnect to the website, if I open the website and turn off the Wi-Fi connection on my phone and reconnect to the Wi-Fi of the Pico. (If i close the website before turning off the Wi-Fi, it works.) 

Im using a Raspberry Pi Pico W in AP mode. 

![image](https://github.com/user-attachments/assets/5f085931-e1e0-414a-9e82-1da116f87b25)

**Comment by miguelgrinberg** [comment]
If the network dies, the Microdot cannot continue working and it has no way to recover. My guess is that when this happens, the `app.run()` or `app.start_server()` calls will raise a network exception. What you need to do when this happens is catch the exception, reconnect to the network, and once the connection is back, start the web server again.

**Reply by tamas9804** [reply]
Thank you for the quick reply. Do i have to shutdown the server before call it again or not?

**Reply by tamas9804** [reply]
I have tried this methode, but it seems like it stucks in /insert-footer-ws while cycle, after shutdown and restart the server.
I have added this code to the previous one:
![image](https://github.com/user-attachments/assets/ef4711ad-1767-4a51-bd17-d1d473637e07)

When I detect the timeout I change a global variable and shut down the server, after the app.start_server come back and wait 1 sec and the server start again. Instead of it, the server still stuck in @app.route('/insert-footer-ws') while cycle and always get timeout and server start again and again. 

![image](https://github.com/user-attachments/assets/64d0ac40-ec55-4248-bcd7-3bca589bd941)

**Reply by miguelgrinberg** [reply]
You need to catch the exception on the `start_server` call. You are catching an error inside a route and trying to do a graceful shutdown, which is not going to work when the network is gone.

```python
while True:
    connect_to_wifi()
    try:
        self.app.run(...)
    except OSError:
        pass
```
